src/model/gateway.rs

```diff
@@ -506,6 +507,41 @@ pub struct SessionStartLimit {
     /// The total number of session starts within the ratelimit period allowed.
     pub total: u64,
 }
+
+#[derive(Debug, Clone, Copy)]
+pub struct ShardInfo {
+    pub id: u32,
+    pub total: u32,
+}
+
+impl ShardInfo {
+    #[must_use]
+    pub fn new(id: u32, total: u32) -> Self {
```

Comment:
Could remove this method entirely. If the goal of ShardInfo was to be clearer on what the first and second number is by naming them, `ShardInfo::new(1, 1)` doesn't help

Reply:
This was mostly made for internal use (where we are passing variables in with the same names anyway), thoughts on changing to pub(crate) instead?

Reply:
`ShardInfo { id, total }` is shorter than

`ShardInfo::new(id, total)` :D

Plus, subjectively, I find it aesthetically pleasing to have few methods on plain-old-data types (structs with only public fields; that are just dumb containers of their data)

Reply:
`ShardInfo::new(1, 1)` could be replaced with a `Default` implementation.
But wouldn't `id=0 total=1` make more sense?

Reply:
`ShardInfo { id, total }` is expanded to this by `cargo +nightly fmt`
![image](https://user-images.githubusercontent.com/45660393/174298692-357c0f63-ac2a-470f-9fd0-18332de8324b.png)
(sorry for image, github markdown was working weird)

Reply:
I'm fine with keeping it for internal use (`pub(crate)`).